sandbox/modules/identity/src/main/java/org/opensearch/identity/IdentityPlugin.java

```diff
@@ -221,4 +265,57 @@ public Collection<Object> createComponents(
 
         return components;
     }
+
+    @Override
+    public Collection<Class<? extends LifecycleComponent>> getGuiceServiceClasses() {
+
+        if (!enabled) {
+            return Collections.emptyList();
+        }
+
+        final List<Class<? extends LifecycleComponent>> services = new ArrayList<>(1);
+        services.add(GuiceHolder.class);
+        return services;
+    }
+
+    public static class GuiceHolder implements LifecycleComponent {
```

Comment:
Since we are in core we should be able to reuse its guice injector directly, not sure how, but I'd hope there is a way!

Reply:
Note; doesn't need to block this from merging IMO

Reply:
Oh really? I will take a look. I am making good progress on the permissions api and hope to get that merged early next week. This will then be next and then I am going to connect to the identity index for permission storage. Thank you for the advice!